test/validators/transfer.ts

```diff
@@ -1,86 +1,144 @@
+      helpers.DataGenerator.pod.byId(podResult2.podId),
+      helpers.DataGenerator.shares(helpers.DB.validators.length),
+      '10000'
+    )).to.be.revertedWith('ValidatorNotOwned');
+  });
+
+  it('Transfer validator to a pod with 7 operators', async () => {
+    // Register validator with 7 operators
+    const { podId } = await helpers.registerValidators(4, 1, '10000', [1, 2, 3, 4, 5, 6, 7]);
+
+    // Transfer validator to an existing pod
+    await expect(ssvNetworkContract.connect(helpers.DB.owners[4]).transferValidator(
+      podResult1.validators[0].publicKey,
+      helpers.DataGenerator.pod.byId(podId),
+      helpers.DataGenerator.shares(helpers.DB.validators.length),
+      '10000'
+    )).to.emit(ssvNetworkContract, 'ValidatorTransferred');
+  });
+
+  // THIS NEEDS SOME PROPER ERROR MESSAGE
+  it('Transfer validator with not enough amount', async () => {
+    // Register validator
+    const { podId } = await helpers.registerValidators(4, 1, '10000', [1, 2, 3, 9]);
+
+    // Increase operator fee
+    await ssvNetworkContract.connect(helpers.DB.owners[7]).updateOperatorFee(9, '100000')
+
+    // Transfer to pod with not enough amount
+    await expect(ssvNetworkContract.connect(helpers.DB.owners[4]).transferValidator(
+      podResult1.validators[0].publicKey,
+      helpers.DataGenerator.pod.byId(podId),
+      helpers.DataGenerator.shares(helpers.DB.validators.length),
+      '1'
+    )).to.be.revertedWith('account liquidatable');
   });
-});
+
+  // THIS NEEDS SOME PROPER ERROR MESSAGE
+  it('Transfer validator with an invalid pod', async () => {
+    await expect(ssvNetworkContract.connect(helpers.DB.owners[4]).transferValidator(
+      podResult1.validators[0].publicKey,
+      podResult2.validators[0].publicKey,
+      helpers.DataGenerator.shares(helpers.DB.validators.length),
+      '10000'
+    )).to.be.revertedWith('InvalidPod');
+  });
+
+  // THIS NEEDS SOME PROPER ERROR MESSAGE
+  it('Transfer validator with not enough balance', async () => {
+    await expect(ssvNetworkContract.connect(helpers.DB.owners[4]).transferValidator(
+      podResult1.validators[0].publicKey,
+      helpers.DataGenerator.pod.byId(podResult2.podId),
+      helpers.DataGenerator.shares(helpers.DB.validators.length),
+      '100001'
+    )).to.be.revertedWith('NotEnoughBalance');
+  });
+
+  // MAYBE WE WILL ADD SOME VALIDITY HERE?
+  // it('Transfer validator with an invalid share', async () => {
+  //   await helpers.registerValidators(4, 1, '10000', helpers.DataGenerator.pod.new());
+  //   const { podId } = await helpers.registerValidators(4, 1, '10000', helpers.DataGenerator.pod.new());
+  //   await expect(ssvNetworkContract.connect(helpers.DB.owners[4]).transferValidator(
+  //     helpers.DataGenerator.publicKey(0),
+  //     helpers.DataGenerator.pod.byId(podId),
+  //     helpers.DataGenerator.shares(helpers.DB.validators.length),
+  //     '10000'
+  //   )).to.be.revertedWith('InvalidShares');
+  // });
```

Comment:
@lior-blox lets decide if we want share validity and if not lets get rid of this test in this PR